### PR TITLE
[irods/irods#4761] Added support for case-insensitive search to ilocate (4-2-stable)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: bin/ilocate [hunk 1]
```diff
@@ -2,8 +2,6 @@
 # ilocate - simple script to emulate functionality of GNU's [m]locate command for iRODS
 # Authored by Scott Sinno, NASA NCCS, scott.s.sinno@nasa.gov
 #
-#Todo : 
-#	Add functionality for case-insensitive search, preferably by way of 
 # adding support for it in iquest.
```
Comment (trel) [inline bin/ilocate:5]: missed a line...

File: bin/ilocate [hunk 2]
```diff
@@ -20,21 +18,29 @@ then
...
-		then
-			echo "$sql_output" | grep -v "CAT_NO_ROWS_FOUND"
-			return 0
-		else
-			return 1
-		fi
+        if [ $case_insensitive -eq 0 ]; then
+		sql_output=`iquest --no-page "%s/%s" "select COLL_NAME, DATA_NAME WHERE $1 $operand '$2'" 2> /dev/null`
+        else
+		sql_output=`iquest uppercase --no-page "%s/%s" "select COLL_NAME, DATA_NAME WHERE $1 $operand '${2^^}'" 2> /dev/null`
```
Comment (trel) [inline bin/ilocate:34]: this requires bash 4+?   please confirm that's shipped with our supported OS/versions.

Comment (korydraughn) [reply bin/ilocate]: replaced the solution with a bash 3 impl. the `awk` impl should be safe.

thoughts?

**Comment by korydraughn** [comment]
Just added `ilocate` to `ihelp`. I think that is all that is needed to make `ilocate` appear in the docs?

File: bin/ilocate (hunk 2 above)
Comment (trel) [reply bin/ilocate]: seems good.

we have a test?

**Comment by trel** [comment]
i believe that's true - the docs are generated from `ihelp -a`

**Comment by korydraughn** [comment]
The formatting for this file is off. I'll expand everything to spaces and add another commit for the same issue.

File: bin/ilocate (hunk 2 above)
Comment (korydraughn) [reply bin/ilocate]: i didn't see any tests for it. i'll look again.

Comment (korydraughn) [reply bin/ilocate]: Added test.

**Comment by korydraughn** [comment]
Added #

**Pull request merged**